### PR TITLE
Add support for SHA-512/224 and SHA-512/256 to SOFT and ICA token

### PR DESCRIPTION
**Pull request by ifranzki**

### CODE REVIEW

File: usr/lib/pkcs11/soft_stdll/soft_specific.c [hunk 1]
```diff
@@ -2268,6 +2340,13 @@ CK_RV token_specific_sha_final(STDLL_TokData_t *tokdata, DIGEST_CONTEXT *ctx,
         return CKR_FUNCTION_FAILED;
     }
 
+    switch (ctx->mech.mechanism) {
+    case CKM_SHA512_224:
+    case CKM_SHA512_256:
+        memcpy(orig_out_data, temp_out_data, hlen);
```
Comment (p-steuer) [inline usr/lib/pkcs11/soft_stdll/soft_specific.c]: You may want to erase the non-output bytes from memory here: OPENSSL_cleanse(temp_out_data, sizeof(temp_out_data))

File: usr/lib/pkcs11/soft_stdll/soft_specific.c [hunk 2]
```diff
@@ -2166,6 +2217,13 @@ CK_RV token_specific_sha(STDLL_TokData_t *tokdata, DIGEST_CONTEXT *ctx,
     if (!rc)
         goto error;
 
+    switch (ctx->mech.mechanism) {
+    case CKM_SHA512_224:
+    case CKM_SHA512_256:
+        memcpy(orig_out_data, temp_out_data, hlen);
```
Comment (p-steuer) [inline usr/lib/pkcs11/soft_stdll/soft_specific.c]: You may want to erase the non-output bytes from memory here: OPENSSL_cleanse(temp_out_data, sizeof(temp_out_data))

File: usr/lib/pkcs11/ica_s390_stdll/ica_specific.c [hunk 3]
```diff
@@ -120,33 +120,76 @@ ica_ec_key_free_t               p_ica_ec_key_free;
...
+        p_ica_ec_key_get_public_key  != NULL &&
+        p_ica_ec_key_get_private_key  != NULL &&
+        p_ica_ec_key_free != NULL)
+        return 1;
+
+    return 0;
+}
+#endif
+
+#ifdef SHA512_224
```
Comment (p-steuer) [inline usr/lib/pkcs11/ica_s390_stdll/ica_specific.c]: where is SHA512_224 etc defined? in ica_api.h ?

File: usr/lib/pkcs11/soft_stdll/soft_specific.c (hunk 1 above)
Comment (ifranzki) [reply usr/lib/pkcs11/soft_stdll/soft_specific.c]: fixed

File: usr/lib/pkcs11/soft_stdll/soft_specific.c (hunk 2 above)
Comment (ifranzki) [reply usr/lib/pkcs11/soft_stdll/soft_specific.c]: fixed

File: usr/lib/pkcs11/ica_s390_stdll/ica_specific.c (hunk 3 above)
Comment (ifranzki) [reply usr/lib/pkcs11/ica_s390_stdll/ica_specific.c]: Yes in ica_api.h . That way I detect an older version libica and build the ICA token without support for SHA-512/224/256.

**Pull request merged**